dotnet/src/Connectors/Connectors.HuggingFace/Core/HuggingFaceClient.cs

```diff
@@ -0,0 +1,209 @@
+    private HttpClient HttpClient { get; }
+    private ILogger Logger { get; }
+
+    internal HuggingFaceClient(
+        string modelId,
+        HttpClient httpClient,
+        IHttpRequestFactory httpRequestFactory,
+        IEndpointProvider endpointProvider,
+        string? apiKey = null,
+        IStreamJsonParser? streamJsonParser = null,
+        ILogger? logger = null)
+    {
+        Verify.NotNullOrWhiteSpace(modelId);
+
+        this._modelId = modelId;
+        this._apiKey = apiKey;
+        this.HttpClient = httpClient;
+        this.HttpRequestFactory = httpRequestFactory;
+        this.EndpointProvider = endpointProvider;
+        this.Logger = logger ?? NullLogger.Instance;
+        this._streamJsonParser = streamJsonParser ?? new TextGenerationStreamJsonParser();
+    }
+
+    public async Task<IReadOnlyList<TextContent>> GenerateTextAsync(string prompt, PromptExecutionSettings? executionSettings, CancellationToken cancellationToken)
+    {
+        var endpoint = this.EndpointProvider.GetTextGenerationEndpoint(executionSettings?.ModelId ?? this._modelId);
+        var request = this.CreateTextRequest(prompt, executionSettings);
+        using var httpRequestMessage = this.HttpRequestFactory.CreatePost(request, endpoint, this._apiKey);
+
+        string body = await this.SendRequestAndGetStringBodyAsync(httpRequestMessage, cancellationToken)
+            .ConfigureAwait(false);
+
+        var response = DeserializeResponse<TextGenerationResponse>(body);
+        var textContents = GetTextContentFromResponse(response, executionSettings?.ModelId ?? this._modelId);
+
+        this.LogTextGenerationUsage(executionSettings);
+
+        return textContents;
+    }
+
+    public async IAsyncEnumerable<StreamingTextContent> StreamGenerateTextAsync(
+        string prompt,
+        PromptExecutionSettings? executionSettings = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var endpoint = this.EndpointProvider.GetTextGenerationEndpoint(executionSettings?.ModelId ?? this._modelId);
+        var request = this.CreateTextRequest(prompt, executionSettings);
+        request.Stream = true;
+
+        using var httpRequestMessage = this.HttpRequestFactory.CreatePost(request, endpoint, this._apiKey);
+
+        using var response = await this.SendRequestAndGetResponseImmediatelyAfterHeadersReadAsync(httpRequestMessage, cancellationToken)
+            .ConfigureAwait(false);
+
+        using var responseStream = await response.Content.ReadAsStreamAndTranslateExceptionAsync()
+            .ConfigureAwait(false);
+
+        foreach (var streamingTextContent in this.ProcessTextResponseStream(responseStream, executionSettings?.ModelId ?? this._modelId))
+        {
+            yield return streamingTextContent;
+        }
+    }
+
+    public async Task<IList<ReadOnlyMemory<float>>> GenerateEmbeddingsAsync(
+        IList<string> data,
+        Kernel? kernel = null,
```

Comment:
1. This method accepts kernel others don't. We need to be consistent.
2. Don't we need to be consistent and get modelId from execution settings as other public methods of the class do?

Reply:
I think the consistency is a good point here, this method is following what we have for the Memory Interface abstraction `IEmbeddingGenerationService`, any changes to this will be breaking changes moving forward or we can think about introducing a new Interface with `PromptExecutionSettings` as a parameter, removing the Kernel or thinking on a better strategy for this.